src/measure_theory/lp_space.lean

```diff
@@ -1785,9 +1787,61 @@ linear_map.mk_continuous
   _
   Lp_norm_le
 
+variables {E p 𝕜}
+
 lemma to_Lp_norm_le [nondiscrete_normed_field 𝕜] [opens_measurable_space 𝕜] [normed_space 𝕜 E]
   [fact (1 ≤ p)] :
   ∥to_Lp E p μ 𝕜∥ ≤ (measure_univ_nnreal μ) ^ (p.to_real)⁻¹ :=
 linear_map.mk_continuous_norm_le _ ((measure_univ_nnreal μ) ^ (p.to_real)⁻¹).coe_nonneg _
 
 end bounded_continuous_function
+
+namespace continuous_map
+
+open_locale bounded_continuous_function
+
+variables [borel_space E] [second_countable_topology E]
+variables [topological_space α] [compact_space α] [borel_space α]
+variables [finite_measure μ]
+
+variables (𝕜 : Type*) [measurable_space 𝕜] (E p μ) [fact (1 ≤ p)]
+
+/-- The bounded linear map of considering a continuous function on a compact finite-measure
+space `α` as an element of `Lp`.  By definition, the norm on `C(α, E)` is the sup-norm, transferred
+from the space `α →ᵇ E` of bounded continuous functions, so this construction is just a matter of
+transferring the structure from `bounded_continuous_function.to_Lp` along the isometry. -/
+def to_Lp [normed_field 𝕜] [opens_measurable_space 𝕜] [normed_space 𝕜 E] :
+  C(α, E) →L[𝕜] (Lp E p μ) :=
+(bounded_continuous_function.to_Lp E p μ 𝕜).comp
+  (linear_isometry_bounded_of_compact α E 𝕜).to_linear_isometry.to_continuous_linear_map
+
+variables {E p 𝕜}
+
+lemma to_Lp_def [normed_field 𝕜] [opens_measurable_space 𝕜] [normed_space 𝕜 E] (f : C(α, E)) :
+  to_Lp E p μ 𝕜 f
+  = bounded_continuous_function.to_Lp E p μ 𝕜 (linear_isometry_bounded_of_compact α E 𝕜 f) :=
+rfl
+
+@[simp] lemma to_Lp_comp_forget_boundedness [normed_field 𝕜] [opens_measurable_space 𝕜]
+  [normed_space 𝕜 E] (f : α →ᵇ E) :
+  to_Lp E p μ 𝕜 (bounded_continuous_function.forget_boundedness α E f)
+  = bounded_continuous_function.to_Lp E p μ 𝕜 f :=
+rfl
+
+@[simp] lemma coe_to_Lp [normed_field 𝕜] [opens_measurable_space 𝕜] [normed_space 𝕜 E]
+  (f : C(α, E)) :
+  (to_Lp E p μ 𝕜 f : α →ₘ[μ] E) = f.to_ae_eq_fun μ :=
+rfl
+
+variables [nonempty α] [nontrivial E] [nondiscrete_normed_field 𝕜] [opens_measurable_space 𝕜]
```

Comment:
Why should `nonempty α` and `nontrivial E` be necessary for the next statements? Otherwise, all norms are zero, so I think the statements are still true.

Reply:
You're right, thanks!  I had used a nontriviality hypothesis in a result in `normed_space/operator_norm`, namely `op_norm_comp_linear_isometry_equiv`, but it also works in the subsingleton case.  Fixed now.